salt/modules/boto_lambda.py

```diff
@@ -0,0 +1,783 @@
+# -*- coding: utf-8 -*-
+'''
+Connection module for Amazon Lambda
+
+.. versionadded:: 
+
+:configuration: This module accepts explicit Lambda credentials but can also
+    utilize IAM roles assigned to the instance trough Instance Profiles.
+    Dynamic credentials are then automatically obtained from AWS API and no
+    further configuration is necessary. More Information available at:
+
+    .. code-block:: text
+
+        http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
+
+    If IAM roles are not used you need to specify them either in a pillar or
+    in the minion's config file:
+
+    .. code-block:: yaml
+
+        lambda.keyid: GKTADJGHEIQSXMKKRBJ08H
+        lambda.key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+
+    A region may also be specified in the configuration:
+
+    .. code-block:: yaml
+
+        lambda.region: us-east-1
+
+    If a region is not specified, the default is us-east-1.
+
+    It's also possible to specify key, keyid and region via a profile, either
+    as a passed in dict, or as a string to pull from pillars or minion config:
+
+    .. code-block:: yaml
+
+        myprofile:
+            keyid: GKTADJGHEIQSXMKKRBJ08H
+            key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+            region: us-east-1
+
+.. versionchanged:: 2015.8.0
+    All methods now return a dictionary. Create and delete methods return:
+
+    .. code-block:: yaml
+
+        created: true
+
+    or
+
+    .. code-block:: yaml
+
+        created: false
+        error:
+          message: error message
+
+    Request methods (e.g., `describe_function`) return:
+
+    .. code-block:: yaml
+
+        function:
+          - {...}
+          - {...}
+
+    or
+
+    .. code-block:: yaml
+
+        error:
+          message: error message
+
+:depends: boto3
+
+'''
+# keep lint from choking on _get_conn and _cache_id
+#pylint: disable=E0602
+
+# Import Python libs
+from __future__ import absolute_import
+import logging
+import socket
+from distutils.version import LooseVersion as _LooseVersion  # pylint: disable=import-error,no-name-in-module
+
+# Import Salt libs
+import salt.utils.boto3
+import salt.utils.compat
+from salt.exceptions import SaltInvocationError, CommandExecutionError
+# from salt.utils import exactly_one
+# TODO: Uncomment this and s/_exactly_one/exactly_one/
+# See note in utils.boto
+
+log = logging.getLogger(__name__)
+
+# Import third party libs
+import salt.ext.six as six
+# pylint: disable=import-error
+try:
+    #pylint: disable=unused-import
+    import boto
+    import boto3
+    #pylint: enable=unused-import
+    from botocore.exceptions import ClientError
+    logging.getLogger('boto').setLevel(logging.CRITICAL)
+    logging.getLogger('boto3').setLevel(logging.CRITICAL)
+    HAS_BOTO = True
```

Comment:
I'm not quite sure what you want to move. This code was lifted from boto_vpc then tweaked for boto3, so  I'm not entirely certain of the reasoning for its current form. I just tried to follow suit from existing code.